### DISCUSSION TITLE
How to create a route with both GET and POST method ?

### DISCUSSION THREAD

**Discussion by pubpy2015** [scottoffen/grapevine#79] [Q&A] [answered]
In Grapevine V4, i can do that:

```csharp
[RestRoute(PathInfo = @"/")]
public IHttpContext ROOT(IHttpContext context)
{
    context.Response.SendResponse(Encoding.UTF8.GetBytes("Welcome to Grapevine"));
    return context;
}
```

**Comment by seikosantana** [comment]
I saw this article: https://scottoffen.github.io/grapevine/docs/routes

`If the method is omitted, the route is assigned the default method HttpMethod.Any, which, as the name suggests, will match any HTTP method.
`
We could use "Any" or HttpMethod.Any enum when specifying the route.
I guess that would work. Since that matches any methods, if you want it to specifically match only GET and POST methods, maybe could do some branching in the method assigned to the route to check the method from the `context`.

Hope it helps.

**Comment by pubpy2015** [comment]
Thank you very much.

**Comment by scottoffen** [comment, marked as answer]
You can also use multiple `Route` attributes on the method.

```csharp
[RestRoute("Get", "/hello")]
[RestRoute("Post", "/hello")]
public async Task HelloWorld(IHttpContext context)
{
    await context.Response.SendResponseAsync($"Hello, world!");
}
```

However, you would have to differentiate in the method if you want the behavior to change depending on whether the method is `POST` or `GET`.

**Comment by pubpy2015** [comment]
Thanks,

An other question:

I am building a server on my PC (IP address of my PC is: 192.168.100.92) as:

`string _serverPort = "1122";`
`var server = RestServerBuilder.UseDefaults().Build();`
`server.Prefixes.Add($"http://localhost:{_serverPort}/");`
`server.Start();`

but when trying access to http://127.0.0.1:1122/api/test or http://192.168.100.92:1122/api/test i get the result is:
![ảnh](https://user-images.githubusercontent.com/59004953/135564614-0d01a026-ee26-4ff1-ba8a-4f4bd9c2d73c.png)

only when access to http://localhost:1122/api/test i get the corect result:
![ảnh](https://user-images.githubusercontent.com/59004953/135564673-16b3a07d-d34b-406a-b643-02d83023dff9.png)

How to solve this problem ?
On your V4, I can do all with:
`server.Host = "+";`
`server.Port = 1122;`
`server.Start();`

**Comment by pubpy2015** [comment]
Oh,
seem replace "localhost" by "+" will solve this problem:
`server.Prefixes.Add($"http://+:{_serverPort}/");`
